src/app/application.cpp

```diff
@@ -278,7 +279,7 @@ void Application::torrentFinished(BitTorrent::TorrentHandle *const torrent)
         program.replace("%T", torrent->currentTracker());
         program.replace("%I", torrent->hash());
 
-        QProcess::startDetached(program);
+        std::thread ([program]() { return system(program.toStdString().c_str()); }).detach();
```

Comment:
>system() will remain waiting, its thread too and qBittorrent process will waiting for this thread.

That not true on linux, please try out the sleep example: https://github.com/qbittorrent/qBittorrent/pull/5296#discussion_r64432775. qbt exits immediately while the `sleep` remains.